In [1]:
from vampyr import vampyr3d as vp
import numpy as np
import matplotlib.pyplot as plt
# from copy import deepcopy

# import KAIN
import utils
# from spinor import spinor
from scfsolv import scfsolv as ss
from scfsolv import scfsolv_1stpert as ss1

prec = 1.0e-4
khist = 4
thrs = prec
c =  137.035999177
# c = 60

N_components = 2
# Pord = 1 #maximum pertubation order computed

# r_x = np.linspace(-16, 16, 10000) # create an evenly spaced set of points between -0.99 and 0.99
# r_y = np.linspace(-16, 16, 10000)
# r_z = np.linspace(-16, 16, 10000)
# r = [r_x, r_y, r_z]
# Restricted = True


# Electric_field = [0.01, 0., 0.]
Electric_field = [0.01, 0., 0.]
# Analytic nuclear potential
# print("pouet")
#H
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [1], "/home/qpitto/Tests_KAIN/test_H/norun/orbitals/", 1, 1, False #Initial guess 
#He 
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [2], "/home/qpitto/Tests_KAIN/test_He/norun/orbitals/", 2, 1, False #Initial guess
R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [2], "/home/qpitto/Tests_KAIN/test_He/run/orbitals/", 2, 2, False #Converged NR result
#H2
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.73699313, 0.100, 0.100], [-0.73699313, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1., 0.100, 0.100], [-1., 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.7, 0.100, 0.100], [-0.7, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.6992, 0.100, 0.100], [-0.6992, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
#H2+ 
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1., 0.100, 0.100], [-1., 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 1, 1, False #Initial guess
# H2plus_dirac = -1.1026415810325771641
# lambda_factor = (2*c**2 + H2plus_dirac)/(2*c**2)
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1/lambda_factor, 0.100, 0.100], [-1/lambda_factor, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 1, 1, False #Initial guess
#LiH
# R, Z, pathMol, nElectron = [[1.508001,  0.188973, 0.188973], [-1.508001,  0.188973, 0.188973]], [3,1], "/home/qpitto/Tests_KAIN/test_LiH/norun/orbitals/", 2 #Initial guess
# R, Z, pathMol, nElectron = [[1.508001,  0.188973, 0.188973], [-1.508001,  0.188973, 0.188973]], [3,1], "/home/qpitto/Tests_KAIN/test_LiH/run/orbitals/", 2 #LiH
#Ne
# R, Z, pathMol, nElectron = [[0.100, 0.100, 0.100]], [10], "/home/qpitto/Tests_KAIN/test_Ne/norun/orbitals/phi_p_scf_", 5 #Ne
#H2O
# R, Z, pathMol, nElectron = [[0.0000,  0.000, -0.125], [-1.4375,  0.000,  1.025], [1.4375,  0.000,  1.025]], [8,1,1], "/home/qpitto/Tests_KAIN/test_water/run/orbitals/", 5 #H2O
# R, Z, pathMol, nElectron = [[0.0000,  0.000, -0.125], [-1.4375,  0.000,  1.025], [1.4375,  0.000,  1.025]], [8,1,1], "/home/qpitto/Tests_KAIN/initial_guesses/phi_p_scf_", 5 #H2O
#CH4
# R, Z, pathMol, nElectron = [[0.1, 0.1, 0.1], [0.1, 0.1, 2.15413215], [2.03631019, 0.1, -0.5856834], [-0.54635362, 1.92524633, -0.5856834], [-0.54635362, -1.72524633, -0.5856834]], [6,1,1,1,1], "/home/qpitto/Tests_KAIN/test_CH4/norun/orbitals/", 5 #CH4

nz = len(Z)

#create scf solver instance with default MRA
Scfs = ss(N_components, prec, khist, scalar=scalar, lightspeed=c)

# print("testLC: ", utils.LeviCivita([1,2,3]), utils.LeviCivita([1,1,3]), utils.LeviCivita([2,1,3]))
# from spinor import spinor
# print("Test spinors")
# spineur = spinor(Scfs.mra, 2)
# spineur.setZero()
# print("length", len(spineur))
# spindeur = spineur + spineur
# print("+ ok")
# spindeur = spineur - spineur
# print("- ok")
# spindeur = 1. * spineur
# spindeur = 2 * spineur
# spindeur = 1j * spineur
# spindeur = spineur * 1
# print("* ok")
# spindeur = spineur / 2
# spindeur = spineur / 2.
# print("norme = ", spineur.compSqNorm())
# print("dot = ", spineur.dot(spindeur))
# spindeur = spineur.derivative(0)
# print("Derivative ok")


# tut = [[-1.514986e+00, -8.879809e-02],
# [-8.879809e-02, 1.188349e+00]]
# print(np.linalg.eig(tut))

#initialise molecule geometry and properties
print("start initMolec")
Scfs.initMolec(nElectron, R, Z, pathMol, InitGuessType, scalar)
print("init done")
#run the SCF solver
Scfs.scfRun(thrs, True, True)

# Scfs_2C = ss(N_components, 0.01*prec, khist, scalar = False)
# reprojected_copy = Scfs.phi_prev[0][-1].reproject(Scfs_2C.P_eps)
# # print("tutbim")
# # Scfs_2C.initMolec(nElectron, R, Z, Scfs, restricted=False, copy_orbitals=True)
# # Scfs_2C.scfRun(thrs, True, True)

# print("Sanity test")
# for i in range(len(Scfs.phi_prev)):
#     for j in range(len(Scfs.phi_prev)):
#         print(f"Test: {i}, {j}", vp.dot(Scfs.phi_prev[i][-1], Scfs.phi_prev[j][-1]))

# print("Start perturbation")
# Scfs_pert = ss1(Scfs)

# print("Contructor passed")
# Scfs_pert.initMolec(Electric_field)
# # Scfs_pert.print_operators()

# Scfs_pert.scfRun(thrs, True, True)

start initMolec
Initmolec
S initial guess =  [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306249392610726+0j)
Expectation value gradk*Nab :  (-0.00012044563871625121+0j)
Expectation value sporb :  1.0530813269418059e-20j
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306249392610726+0j)
Expectation value gradk*Nab :  (-0.00012044563871625121+0j)
Expectation value sporb :  1.0530813269418059e-20j
Fock [[-0.75587279-9.16960004e-21j]]
Kinetic [[1.43039976-7.60688169e-21j]]
Vnuc [[-3.21193967+0.j]]
Coulomb [[2.05133425+0.j]]
Exchange [[1.02566713+0.j]]


/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:170: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:650: ComplexWarning: Casting complex values to real discards the imaginary part
  VzFphi = V_z/(self.c**2) * self.Fock[orb, orb] * self.phi_prev[orb][-1]


Power Iter, norm of new guess =  (1.1636000221646157+0j) (1.1636000220968359+0j) (6.777983127744903e-11+0j)
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-0.75587279-9.16960004e-21j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.234302644185566+1.2164421130315898e-20j)
Expectation value gradk*Nab :  (-0.0001896295500561754+9.27829789425612e-23j)
Expectation value sporb :  (-1.2258840936673134e-09-7.380082494002417e-10j)


/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:505: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)


Power Iter, norm of new guess =  (0.9545144369677607+0j) (0.9545144369532508+0j) (1.450994868789787e-11+0j)
orthonorm
S durant Ortho [[0.95451444+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3737119187150915-2.8933942428536424e-14j)
Expectation value gradk*Nab :  (-0.00024151256882254277-4.445931270218589e-14j)
Expectation value sporb :  (2.560565634338832e-10-1.4171918279845763e-10j)
Fock [[-0.92590432+1.18484902e-09j]]
Kinetic [[1.37335916+1.41568302e-10j]]
Vnuc [[-3.30746259-3.63547091e-22j]]
Coulomb [[2.01639822-3.38319294e-22j]]
Exchange [[1.00819911-1.04328071e-09j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3737119187150915+2.8933942428536424e-14j)
Expectation value gradk*Nab :  (-0.00024151256882254277+4.445931270218589e-14j)
Expectation value sporb :  (3.2221020384703766e-10-1.26383911843977e-10j)
Fock [[-0.92590432-9.16745921e-10j]]
Kinetic [[1.37335916+1.26534792e-10j]]
Vnuc [[-3.30746259+3.63547091e-22j]]
Cou

In [2]:
# Scfs_2C = ss(N_components, 0.01*prec, khist, scalar = False)
# # reprojected_copy = Scfs.phi_prev[0][-1].reproject(Scfs_2C.P_eps)
# print("tutbim")
# Scfs_2C.initMolec(nElectron, R, Z, Scfs, restricted=False, copy_orbitals=True)
# print("===============Init done=============================================")
# Scfs_2C.scfRun(thrs, True, True)

In [3]:
import numpy as np
# F= [[-2.7896489 -2.66352263e-07j, 0.00707083+5.25880281e-08j],[ 0.00704288-1.01844084e-07j, -1.11828297+1.44609334e-08j]] #initial
F=np.array([[-2.7263095 -9.37847108e-05j, -0.04998474+1.23454107e-03j],
 [-0.05001107-9.24036599e-04j, -1.19366291-3.31073015e-06j]]) #final (no kain)
eigvals, U = np.linalg.eig(F)
print(eigvals)

[-2.72793954-8.35624316e-05j -1.19203287-1.35330094e-05j]


In [4]:
# # print("Scaled energy (4C Dirac energy)", (1/(-0.50056707) - 1/(c**2))**(-1))
# # mp = 1.672621925e-27 # in kg
# # me = 9.1093837139e-31 #in kg, but we don't need it 
# me = 1 #in A.U. by definition
# mp = 1.672621925e-27/9.1093837139e-31 * me 
# mu = me*mp/(me+mp)  #TODO convertir en unités atomiques
# print(mu)
# alpha = 1/137.035999177
# n,k = 1,-1 #quantum numbers
# c_original = 1/alpha
# gse_H = mu*c_original**2 / np.sqrt(1 + (Scfs.Z[0]*alpha/(n-np.abs(k) + np.sqrt(k**2 - (Scfs.Z[0]*alpha)**2)))**2)
# # print("Dirac ground state energy of Hydrogen",-13.598433/27.211386245988) #On numerator: Ionisation energy of H in eV (source: NIST), denominator: conversion factor
# print("Dirac ground state energy of Hydrogen",gse_H - c_original**2) #On numerator: Ionisation energy of H in eV (source: Wikipedia, Bohr-Sommerfeld model), denominator: conversion factor
print("Ground state of He in atomic units:", -79.02/27.211386245988)
print(2*1.33203403 + 2*(-3.37472204) + 2.05163792 - 1.02581896)
print(-1.36275874*2)

Ground state of He in atomic units: -2.9039314383202575
-3.05955706
-2.72551748


In [5]:
#From Luca's orbital4c class
def analytic_1s(light_speed, n, k, Z):
    alpha = 1/light_speed
    gamma = compute_gamma(k,Z,alpha)
    tmp1 = n - np.abs(k) + gamma
    tmp2 = Z * alpha / tmp1
    tmp3 = 1 + tmp2**2
    return light_speed**2 / np.sqrt(tmp3)

def compute_gamma(k,Z,alpha):
    return np.sqrt(k**2 - Z**2 * alpha**2)

c_in = c
n,k = 1,-1 #quantum numbers
# c_in = c_original
gse_H_alt = analytic_1s(c_in, n, k, Scfs.Z[0])
# gse_H_alt = analytic_1s(c_in, n, k, 1)
print(gse_H_alt - c_in**2)

-2.0001065140531864


In [6]:
Energy_test = -0.50251687
# print((1/Energy_test - 1/(2*137.035999177**2))**(-1))
print((1/Energy_test - 1/(2*10**2))**(-1))

-0.5012574184550684


H2+ 1e-5

start initMolec
Initmolec
Vnuc init 0
Vnuc init 1
paired idx 0
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.1886664906107738+0j)
Fock [[-1.05635573-1.97881969e-17j]]
Kinetic [[0.3836732-1.97920835e-17j]]
Vnuc [[-1.44002892+0.j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[0.+0.j]]
pouet
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.1886664906107738+0j)
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:188: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
orthonorm
Post-Ortho S= [[1.+0.j]]
0  length update history:  1
Init molec S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-1.05635573-1.97881969e-17j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2370319393834666+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.09078707-5.03400341e-16j]
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:493: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580872825977225+0j)
Fock [[-1.10719273-4.19117521e-16j]]
Kinetic [[0.46768901-1.43728845e-13j]]
Vnuc [[-1.71241118+3.32329486e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[2.07569645e-16+4.11856231e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2582856934314086+0j)
Fock [[-1.10719131-1.16908548e-15j]]
Kinetic [[0.46768923+7.22546442e-14j]]
Vnuc [[-1.71241118-3.32329486e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[2.07569645e-16-4.11856231e-11j]]
Orbital: 0    Norm: [1.04247606]    Update: [0.1528374]    Energy:[-1.09078707-5.03400341e-16j]
Scaled energy (4C Dirac energy) [-1.08487026]
=============Iteration: 1
F= [[-1.10719131-1.16908548e-15j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2571252068384249+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.10718998-3.85567612e-16j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580252850153615+0j)
Fock [[-1.10727626-6.62121785e-16j]]
Kinetic [[0.46598752-1.20061441e-13j]]
Vnuc [[-1.70695937-6.07036539e-21j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[2.43251012e-16+5.60785411e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2582398254823342+0j)
Fock [[-1.10727476-9.09868696e-16j]]
Kinetic [[0.46598775+5.4156487e-14j]]
Vnuc [[-1.70695937+6.07036539e-21j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[2.43251012e-16-5.60785411e-11j]]
Orbital: 0    Norm: [1.00702622]    Update: [0.01170772]    Energy:[-1.10718998-3.85567612e-16j]
Scaled energy (4C Dirac energy) [-1.10109438]
=============Iteration: 2
F= [[-1.10727476-9.09868696e-16j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2570335119900062+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.10727351-6.01140422e-16j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580269610277435+0j)
Fock [[-1.10731399-7.41739055e-16j]]
Kinetic [[0.46720426-3.18435937e-14j]]
Vnuc [[-1.71126654+9.60467079e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.02933438e-16+7.90196948e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.258271478872725+0j)
Fock [[-1.1073123-8.45086495e-16j]]
Kinetic [[0.46720452-3.162933e-14j]]
Vnuc [[-1.71126654-9.60467079e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.02933438e-16-7.90196948e-11j]]
Orbital: 0    Norm: [1.00121131]    Update: [0.0038407]    Energy:[-1.10727351-6.01140422e-16j]
Scaled energy (4C Dirac energy) [-1.10117699]
=============Iteration: 3
F= [[-1.1073123-8.45086495e-16j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.256955564982991+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.10731119-5.45387406e-16j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580165348162131+0j)
Fock [[-1.10731289-7.4448408e-16j]]
Kinetic [[0.46694037-1.58315259e-13j]]
Vnuc [[-1.71032663-1.2001001e-19j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.00157847e-16+7.50775912e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2582574914316813+0j)
Fock [[-1.10731121-8.43695109e-16j]]
Kinetic [[0.46694062+9.4815811e-14j]]
Vnuc [[-1.71032663+1.2001001e-19j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.00157847e-16-7.50775912e-11j]]
Orbital: 0    Norm: [0.99997718]    Update: [0.0009362]    Energy:[-1.10731119-5.45387406e-16j]
Scaled energy (4C Dirac energy) [-1.10121426]
=============Iteration: 4
F= [[-1.10731121-8.43695109e-16j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.256949837449211+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.1073101-5.42981529e-16j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580196526500642+0j)
Fock [[-1.10731351-7.4550489e-16j]]
Kinetic [[0.46699261-2.46848116e-13j]]
Vnuc [[-1.71046974-1.64479686e-19j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.03761482e-16+6.80611393e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2582601447172639+0j)
Fock [[-1.10731183-8.46580356e-16j]]
Kinetic [[0.46699287+1.83342647e-13j]]
Vnuc [[-1.71046974+1.64479686e-19j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.03761482e-16-6.80611393e-11j]]
Orbital: 0    Norm: [0.9999996]    Update: [0.00012802]    Energy:[-1.1073101-5.42981529e-16j]
Scaled energy (4C Dirac energy) [-1.10121318]
=============Iteration: 5
F= [[-1.10731183-8.46580356e-16j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2569513360814184+0j)
ExpandSol start S= [[1.+0.j]]
Shell energy [-1.10731073-5.42527261e-16j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.2580194874533692+0j)
Fock [[-1.10731352-7.47108731e-16j]]
Kinetic [[0.46699328-3.14730723e-13j]]
Vnuc [[-1.71046986+6.33236434e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.04694506e-16+6.38155148e-11j]]
end ExpandSol
compute Coulomb onlz 1 e
compFock main loop 0
compute Korb only 1 e
kappa compo 0
kappa compo 1
kappa after
================kappadone==================
compFop kappa done
compFop NabkNab done
compFop sporb done
CompFop real 0 (1.258260010236542+0j)
Fock [[-1.10731184-8.47529827e-16j]]
Kinetic [[0.46699354+2.51287031e-13j]]
Vnuc [[-1.71046986-6.33236434e-20j]]
Coulomb [[0.+0.j]]
Exchange [[0.+0.j]]
Kramer pair overlap [[3.04694506e-16-6.38155148e-11j]]
Orbital: 0    Norm: [1.]    Update: [6.74159032e-06]    Energy:[-1.10731073-5.42527261e-16j]
Scaled energy (4C Dirac energy) [-1.1012138]

He 1e-4 - rho corrected, small issue with kinetic

start initMolec
Initmolec
S initial guess =  [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
compFock main loop 0
compute Korb only 1 e
Fock [[-0.75587279-9.16960004e-21j]]
Kinetic [[1.43039976-7.60688169e-21j]]
Vnuc [[-3.21193967+0.j]]
Coulomb [[2.05133425+0.j]]
Exchange [[1.02566713+0.j]]
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:170: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:624: ComplexWarning: Casting complex values to real discards the imaginary part
  VzFphi = V_z/(self.c**2) * self.Fock[orb, orb] * self.phi_prev[orb][-1]
Power Iter, norm of new guess =  (1.1636000221646157+0j) (1.1636000220968359+0j) (6.777983127744903e-11+0j)
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-0.75587279-9.16960004e-21j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:479: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)
Power Iter, norm of new guess =  (0.9545144369677607+0j) (0.9545144369532508+0j) (1.450994868789787e-11+0j)
orthonorm
S durant Ortho [[0.95451444+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.92590432+1.18484902e-09j]]
Kinetic [[1.31268355-1.77870104e-09j]]
Vnuc [[-3.30746259-3.63547091e-22j]]
Coulomb [[2.01639822-3.38319294e-22j]]
Exchange [[1.00819911-1.04328071e-09j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.92590432-9.16745921e-10j]]
Kinetic [[1.31268355+1.45571224e-09j]]
Vnuc [[-3.30746259+3.63547091e-22j]]
Coulomb [[2.01639822+3.38319294e-22j]]
Exchange [[1.00819911+1.04328071e-09j]]
Orbital: 0    Norm: [0.98572993]    Update: [0.06118879]    Energy:[-0.93510056+7.01888952e-10j]
Scaled energy (4C Dirac energy) [-0.93507728]
=============Iteration: 1
F= [[-0.92590432-9.16745921e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Power Iter, norm of new guess =  (0.9839430195371669+0j) (0.983943019534341+0j) (2.825813499590086e-12+0j)
orthonorm
S durant Ortho [[0.98394302+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91819241-5.15848778e-10j]]
Kinetic [[1.33013424+9.79522462e-10j]]
Vnuc [[-3.37092391-5.68685157e-23j]]
Coulomb [[2.05081339-7.61007173e-23j]]
Exchange [[1.02540669+5.10056896e-10j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91819241+5.08262278e-10j]]
Kinetic [[1.33013424-9.89001161e-10j]]
Vnuc [[-3.37092391+5.68685157e-23j]]
Coulomb [[2.05081339+7.61007173e-23j]]
Exchange [[1.02540669-5.10056896e-10j]]
Orbital: 0    Norm: [1.00145833]    Update: [0.01703904]    Energy:[-0.92599425+1.183405e-09j]
Scaled energy (4C Dirac energy) [-0.92597142]
=============Iteration: 2
F= [[-0.91819241+5.08262278e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Power Iter, norm of new guess =  (0.9989037363648869+0j) (0.998903736364657+0j) (2.2998198792005203e-13+0j)
orthonorm
S durant Ortho [[0.99890374+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91815843+6.88959159e-10j]]
Kinetic [[1.33172824-4.95553041e-10j]]
Vnuc [[-3.37359365+7.65142698e-24j]]
Coulomb [[2.05098455-1.15805199e-23j]]
Exchange [[1.02549228-6.82393467e-10j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91815843-6.76995529e-10j]]
Kinetic [[1.33172824+4.84539687e-10j]]
Vnuc [[-3.37359365-7.65142698e-24j]]
Coulomb [[2.05098455+1.15805199e-23j]]
Exchange [[1.02549228+6.82393467e-10j]]
Orbital: 0    Norm: [0.99947064]    Update: [0.00121474]    Energy:[-0.91828927-5.15871215e-10j]
Scaled energy (4C Dirac energy) [-0.91826682]
=============Iteration: 3
F= [[-0.91815843-6.76995529e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Power Iter, norm of new guess =  (0.9995344453686966+0j) (0.9995344453686966+0j) 0j
orthonorm
S durant Ortho [[0.99953445+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91802105+4.87722836e-11j]]
Kinetic [[1.33198222+1.85282952e-11j]]
Vnuc [[-3.37453882+8.03189155e-24j]]
Coulomb [[2.05153523+5.12464239e-24j]]
Exchange [[1.02576762-4.92489366e-11j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91802105-4.87144622e-11j]]
Kinetic [[1.33198222-1.83815136e-11j]]
Vnuc [[-3.37453882-8.03189155e-24j]]
Coulomb [[2.05153523-5.12464239e-24j]]
Exchange [[1.02576762+4.92489366e-11j]]
Orbital: 0    Norm: [0.99999817]    Update: [0.00022811]    Energy:[-0.91825616+6.88993557e-10j]
Scaled energy (4C Dirac energy) [-0.91823371]
=============Iteration: 4
F= [[-0.91802105-4.87144622e-11j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Power Iter, norm of new guess =  (0.9997722427554468+0j) (0.9997722427554468+0j) 0j
orthonorm
S durant Ortho [[0.99977224+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91799539-1.71548646e-11j]]
Kinetic [[1.33203407+4.07655629e-11j]]
Vnuc [[-3.37472214-4.91138482e-25j]]
Coulomb [[2.05163799-6.62390726e-25j]]
Exchange [[1.025819+1.73006253e-11j]]
compFock main loop 0
compute Korb only 1 e
Fock [[-0.91799539+1.73506028e-11j]]
Kinetic [[1.33203407-4.09972332e-11j]]
Vnuc [[-3.37472214+4.91138482e-25j]]
Coulomb [[2.05163799+6.62390726e-25j]]
Exchange [[1.025819-1.73006253e-11j]]
Orbital: 0    Norm: [0.99999996]    Update: [4.20171306e-05]    Energy:[-0.91811896+4.87696427e-11j]
Scaled energy (4C Dirac energy) [-0.91809652]

start initMolec
Initmolec
S initial guess =  [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4307384790280349+0j)
Expectation value gradk*Nab :  (-0.00015229405557775014+0j)
Expectation value sporb :  1.2394690330517829e-20j
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4307384790280349+0j)
Expectation value gradk*Nab :  (-0.00015229405557775014+0j)
Expectation value sporb :  1.2394690330517829e-20j
Fock [[-0.75595404-1.18906408e-20j]]
Kinetic [[1.43031807-1.35656889e-20j]]
Vnuc [[-3.21193969+0.j]]
Coulomb [[2.05133517+0.j]]
Exchange [[1.02566758+0.j]]
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:170: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:650: ComplexWarning: Casting complex values to real discards the imaginary part
  VzFphi = V_z/(self.c**2) * self.Fock[orb, orb] * self.phi_prev[orb][-1]
Power Iter, norm of new guess =  (1.1634530094260784+0j) (1.16345300935799+0j) (6.808849427497362e-11+0j)
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-0.75595404-1.18906408e-20j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.234608074089371+4.129406170498396e-21j)
Expectation value gradk*Nab :  (-0.00021858185613585432+1.143638599413326e-22j)
Expectation value sporb :  (-1.236938221012683e-09-7.38368057237892e-10j)
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:505: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)
Power Iter, norm of new guess =  (0.9545092003590998+0j) (0.9545092003445679+0j) (1.4531955904847523e-11+0j)
orthonorm
S durant Ortho [[0.9545092+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3740546013699437-9.714133870664012e-14j)
Expectation value gradk*Nab :  (-0.000275598735276108-4.702596643165769e-14j)
Expectation value sporb :  (2.536243914619259e-10-1.533547466803844e-10j)
Fock [[-0.9259426+1.23235915e-09j]]
Kinetic [[1.37351405+1.53115157e-10j]]
Vnuc [[-3.30773613-1.89837749e-22j]]
Coulomb [[2.01655893+5.32290724e-22j]]
Exchange [[1.00827947-1.07924399e-09j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3740546013699437+9.714133870664012e-14j)
Expectation value gradk*Nab :  (-0.000275598735276108+4.702596643165769e-14j)
Expectation value sporb :  (3.5449088455750934e-10-1.2963450494711998e-10j)
Fock [[-0.9259426-9.49369896e-10j]]
Kinetic [[1.37351405+1.29874095e-10j]]
Vnuc [[-3.30773613+1.89837749e-22j]]
Coulomb [[2.01655893-5.32290724e-22j]]
Exchange [[1.00827947+1.07924399e-09j]]
Orbital: 0    Norm: [0.98572375]    Update: [0.06120476]    Energy:[-0.93509881+7.02013256e-10j]
Scaled energy (4C Dirac energy) [-0.93507553]
=============Iteration: 1
F= [[-0.9259426-9.49369896e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3740449881660748-8.724072664697411e-14j)
Expectation value gradk*Nab :  (-0.000275280615769607-4.738897286912419e-14j)
Expectation value sporb :  (2.537197384675201e-10-1.5320745650828464e-10j)
Power Iter, norm of new guess =  (0.9839927811220454+0j) (0.983992781119209+0j) (2.8363432634972416e-12+0j)
orthonorm
S durant Ortho [[0.98399278+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4279890305577545-1.3228940562171242e-12j)
Expectation value gradk*Nab :  (-0.0002962411453948837+4.4582084524649226e-13j)
Expectation value sporb :  (-1.5230192214265818e-11-3.406163688733345e-12j)
Fock [[-0.91824314-4.7255454e-10j]]
Kinetic [[1.4274096+2.47621727e-12j]]
Vnuc [[-3.37111319+1.22629879e-22j]]
Coulomb [[2.05092089-2.32642275e-23j]]
Exchange [[1.02546044+4.75030758e-10j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4279890305577545+1.3228940562171242e-12j)
Expectation value gradk*Nab :  (-0.0002962411453948837-4.4582084524649226e-13j)
Expectation value sporb :  (1.3799796680562877e-11+4.012841418110641e-12j)
Fock [[-0.91824314+4.71947863e-10j]]
Kinetic [[1.4274096-3.082895e-12j]]
Vnuc [[-3.37111319-1.22629879e-22j]]
Coulomb [[2.05092089+2.32642275e-23j]]
Exchange [[1.02546044-4.75030758e-10j]]
Orbital: 0    Norm: [1.00145203]    Update: [0.01701388]    Energy:[-0.92595167+1.23222115e-09j]
Scaled energy (4C Dirac energy) [-0.92592885]
=============Iteration: 2
F= [[-0.91824314+4.71947863e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4279788721133688-1.355235009145226e-12j)
Expectation value gradk*Nab :  (-0.00029588548398110434+4.465074235190115e-13j)
Expectation value sporb :  (-1.51673160233469e-11-3.392863971661317e-12j)
Power Iter, norm of new guess =  (0.9989418643604405+0j) (0.9989418643602069+0j) (2.33601635958024e-13+0j)
orthonorm
S durant Ortho [[0.99894186+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306087818001747+2.095450367341538e-13j)
Expectation value gradk*Nab :  (-0.00029812232676532357-1.1700969041134488e-13j)
Expectation value sporb :  (2.0080035962037788e-11+1.8140729193826387e-12j)
Fock [[-0.91820919+6.87375406e-10j]]
Kinetic [[1.43002597-1.7265882e-12j]]
Vnuc [[-3.37378025-8.14772746e-23j]]
Coulomb [[2.05109018+8.89219087e-23j]]
Exchange [[1.02554509-6.89101994e-10j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306087818001747-2.095450367341538e-13j)
Expectation value gradk*Nab :  (-0.00029812232676532357+1.1700969041134488e-13j)
Expectation value sporb :  (-1.1472102126050178e-11-6.095013109006258e-12j)
Fock [[-0.91820919-6.83094466e-10j]]
Kinetic [[1.43002597+6.00752839e-12j]]
Vnuc [[-3.37378025+8.14772746e-23j]]
Coulomb [[2.05109018-8.89219087e-23j]]
Exchange [[1.02554509+6.89101994e-10j]]
Orbital: 0    Norm: [0.99947274]    Update: [0.001214]    Energy:[-0.91825269-4.72599013e-10j]
Scaled energy (4C Dirac energy) [-0.91823024]
=============Iteration: 3
F= [[-0.91820919-6.83094466e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4305977317024159+2.1384869631726877e-13j)
Expectation value gradk*Nab :  (-0.00029776337624964433-1.1722056713358705e-13j)
Expectation value sporb :  (2.0079749632165297e-11+1.866227144356162e-12j)
Power Iter, norm of new guess =  (0.9995724509231788+0j) (0.9995724509231699+0j) (8.899401252327693e-15+0j)
orthonorm
S durant Ortho [[0.99957245+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.431416231386316-4.1281406886846884e-12j)
Expectation value gradk*Nab :  (-0.00029844397296779505+1.1666134992063302e-13j)
Expectation value sporb :  (-3.6267318556413634e-12-2.971915278255169e-12j)
Fock [[-0.91807157+9.06914612e-11j]]
Kinetic [[1.43083283-1.0474333e-12j]]
Vnuc [[-3.37472448+1.33641369e-23j]]
Coulomb [[2.05164017-3.15362605e-24j]]
Exchange [[1.02582009-9.17388945e-11j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.431416231386316+4.1281406886846884e-12j)
Expectation value gradk*Nab :  (-0.00029844397296779505-1.1666134992063302e-13j)
Expectation value sporb :  (9.111570354162838e-12+2.2117357396895294e-13j)
Fock [[-0.91807157-8.79407195e-11j]]
Kinetic [[1.43083283+3.798175e-12j]]
Vnuc [[-3.37472448-1.33641369e-23j]]
Coulomb [[2.05164017+3.15362605e-24j]]
Exchange [[1.02582009+9.17388945e-11j]]
Orbital: 0    Norm: [0.99999817]    Update: [0.00022778]    Energy:[-0.91821963+6.87327194e-10j]
Scaled energy (4C Dirac energy) [-0.91819718]
=============Iteration: 4
F= [[-0.91807157-8.79407195e-11j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4314047605044977-4.132151505613275e-12j)
Expectation value gradk*Nab :  (-0.00029808387663162226+1.1681408760016155e-13j)
Expectation value sporb :  (-3.51626604765758e-12-3.0041027938274423e-12j)
Power Iter, norm of new guess =  (0.9998102537427339+0j) (0.9998102537427339+0j) 0j
orthonorm
S durant Ortho [[0.99981025+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4315746309282307-5.432711995865248e-13j)
Expectation value gradk*Nab :  (-0.00029851390428082246+6.907916050565115e-14j)
Expectation value sporb :  (1.755412742690283e-12+1.2510813591047202e-12j)
Fock [[-0.91804506-9.17042109e-11j]]
Kinetic [[1.43099109-1.73240421e-12j]]
Vnuc [[-3.37490753-8.78879364e-25j]]
Coulomb [[2.05174275-2.89513224e-24j]]
Exchange [[1.02587138+8.99718067e-11j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4315746309282307+5.432711995865248e-13j)
Expectation value gradk*Nab :  (-0.00029851390428082246-6.907916050565115e-14j)
Expectation value sporb :  (-3.911700831725781e-12-1.6942213309649923e-13j)
Fock [[-0.91804506+9.06225517e-11j]]
Kinetic [[1.43099109+6.50744985e-13j]]
Vnuc [[-3.37490753+8.78879364e-25j]]
Coulomb [[2.05174275+2.89513224e-24j]]
Exchange [[1.02587138-8.99718067e-11j]]
Orbital: 0    Norm: [0.99999996]    Update: [4.19463139e-05]    Energy:[-0.91808242+9.07199022e-11j]
Scaled energy (4C Dirac energy) [-0.91805997]
=============Iteration: 5
F= [[-0.91804506+9.06225517e-11j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4315622638845622-5.461994373370307e-13j)
Expectation value gradk*Nab :  (-0.0002981533457624992+6.918075688591689e-14j)
Expectation value sporb :  (1.7455803470993098e-12+1.2502809804110866e-12j)
Power Iter, norm of new guess =  (0.9998562998282299+0j) (0.9998562998282299+0j) 0j
orthonorm
S durant Ortho [[0.9998563+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.431585659260859-5.756794890258863e-14j)
Expectation value gradk*Nab :  (-0.00029851903912950136+8.806630974244271e-14j)
Expectation value sporb :  (-1.5326022341025922e-14-3.194772028220866e-14j)
Fock [[-0.91804364-1.41998844e-11j]]
Kinetic [[1.43100211+3.24720416e-14j]]
Vnuc [[-3.37491979+3.87740912e-26j]]
Coulomb [[2.05174807+1.29246971e-26j]]
Exchange [[1.02587404+1.42323564e-11j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.431585659260859+5.756794890258863e-14j)
Expectation value gradk*Nab :  (-0.00029851903912950136-8.806630974244271e-14j)
Expectation value sporb :  (8.359674536402595e-14+7.204414945416431e-16j)
Fock [[-0.91804364+1.42311117e-11j]]
Kinetic [[1.43100211-1.24476283e-15j]]
Vnuc [[-3.37491979-3.87740912e-26j]]
Coulomb [[2.05174807-1.29246971e-26j]]
Exchange [[1.02587404-1.42323564e-11j]]
Orbital: 0    Norm: [1.]    Update: [2.45565706e-06]    Energy:[-0.91805679-9.17061616e-11j]
Scaled energy (4C Dirac energy) [-0.91803435]

c = 33
z=10

-0.9184789999999998
0.9994556794244512
Dirac ground state energy of Hydrogen -60.261212302269996
Ground state of He in atomic units: -2.9039314383202575
[-2.72793954-8.35624316e-05j -1.19203287-1.35330094e-05j]
start initMolec
Initmolec
Vnuc init 0
Vnuc init 1
paired idx 0
pouet
orb1 Is rho zero? (should be) True True
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:182: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
orthonorm
Post-Ortho S= [[1.+0.j]]
0  length update history:  1
Init molec S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-8.48629129+4.49944059e-18j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) True True
ExpandSol start S= [[1.+0.j]]
Shell energy [-16.8183771-9.59671403e-13j]
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:471: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-32.92109491-5.00602967e-11j]]
Kinetic [[0.72766051-2.67370336e-13j]]
Vnuc [[-48.44275797-9.30101034e-20j]]
Coulomb [[5.75311272-1.19830431e-20j]]
Exchange [[2.87655636+5.15474322e-11j]]
Kramer pair overlap [[-4.51339076e-11-4.46930356e-11j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-32.92109495-2.21645322e-09j]]
Kinetic [[0.72766051+1.44971012e-10j]]
Vnuc [[-48.44275797+9.30101034e-20j]]
Coulomb [[5.75311272+1.19830431e-20j]]
Exchange [[2.87655636-5.15474322e-11j]]
Kramer pair overlap [[-4.51339076e-11+4.46930356e-11j]]
Orbital: 0    Norm: [1.27111068]    Update: [1.21124672]    Energy:[-16.8183771-9.59671403e-13j]
Scaled energy (4C Dirac energy) [-16.56258663]
=============Iteration: 1
F= [[-32.92109495-2.21645322e-09j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-32.92109558-5.00640424e-11j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-37.67861117-8.49188867e-11j]]
Kinetic [[0.83027672-1.08172863e-12j]]
Vnuc [[-57.08408568-5.9537966e-20j]]
Coulomb [[6.93711417-2.32901668e-21j]]
Exchange [[3.46855708+9.11045902e-11j]]
Kramer pair overlap [[-7.90852891e-11-7.77941009e-11j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-37.67861123-3.82742178e-09j]]
Kinetic [[0.83027672+1.79948355e-10j]]
Vnuc [[-57.08408568+5.9537966e-20j]]
Coulomb [[6.93711417+2.32901668e-21j]]
Exchange [[3.46855708-9.11045902e-11j]]
Kramer pair overlap [[-7.90852891e-11+7.77941009e-11j]]
Orbital: 0    Norm: [0.94718695]    Update: [0.41242024]    Energy:[-32.92109558-5.00640424e-11j]
Scaled energy (4C Dirac energy) [-31.9550753]
=============Iteration: 2
F= [[-37.67861123-3.82742178e-09j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-37.67861352-8.47029169e-11j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-43.48885788-3.25933778e-10j]]
Kinetic [[0.71151838-3.50303667e-12j]]
Vnuc [[-83.72306365-8.58168866e-20j]]
Coulomb [[10.23800621+6.78503815e-20j]]
Exchange [[5.1190031+3.45562577e-10j]]
Kramer pair overlap [[-2.85441293e-10-2.76816078e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-43.48885804-9.19534557e-09j]]
Kinetic [[0.71151838+2.94464525e-10j]]
Vnuc [[-83.72306365+8.58168866e-20j]]
Coulomb [[10.23800621-6.78503815e-20j]]
Exchange [[5.1190031-3.45562577e-10j]]
Kramer pair overlap [[-2.85441293e-10+2.76816078e-10j]]
Orbital: 0    Norm: [0.97814699]    Update: [0.36281449]    Energy:[-37.67861352-8.47029169e-11j]
Scaled energy (4C Dirac energy) [-36.41855772]
=============Iteration: 3
F= [[-43.48885804-9.19534557e-09j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-43.48886064-3.25420357e-10j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-44.97218598-1.15293264e-10j]]
Kinetic [[0.63549107-8.29398126e-12j]]
Vnuc [[-93.96474632+2.75539239e-19j]]
Coulomb [[11.88948717-6.51606358e-21j]]
Exchange [[5.94474358+1.43787601e-10j]]
Kramer pair overlap [[-4.15908822e-10-4.01344821e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-44.97218619-1.26343472e-08j]]
Kinetic [[0.63549107+3.50927953e-10j]]
Vnuc [[-93.96474632-2.75539239e-19j]]
Coulomb [[11.88948717+6.51606358e-21j]]
Exchange [[5.94474358-1.43787601e-10j]]
Kramer pair overlap [[-4.15908822e-10+4.01344821e-10j]]
Orbital: 0    Norm: [1.01710164]    Update: [0.2328178]    Energy:[-43.48886064-3.25420357e-10j]
Scaled energy (4C Dirac energy) [-41.81883891]
=============Iteration: 4
F= [[-44.97218619-1.26343472e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-44.9721899-1.15286166e-10j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.00976739+3.3624126e-10j]]
Kinetic [[0.63292246-1.91423437e-11j]]
Vnuc [[-98.58819636-7.12038726e-19j]]
Coulomb [[12.43119438+2.32335389e-20j]]
Exchange [[6.21559718-2.99164856e-10j]]
Kramer pair overlap [[-5.18911494e-10-4.95764269e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.00976765-1.53487472e-08j]]
Kinetic [[0.63292246+3.95254143e-10j]]
Vnuc [[-98.58819636+7.12038726e-19j]]
Coulomb [[12.43119438-2.32335389e-20j]]
Exchange [[6.21559718+2.99164856e-10j]]
Kramer pair overlap [[-5.18911494e-10+4.95764269e-10j]]
Orbital: 0    Norm: [1.00390247]    Update: [0.05256559]    Energy:[-44.9721899-1.15286166e-10j]
Scaled energy (4C Dirac energy) [-43.18863834]
=============Iteration: 5
F= [[-45.00976765-1.53487472e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-45.00976846+3.50481306e-10j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.07120683+5.30601216e-10j]]
Kinetic [[0.64300226-2.49969731e-11j]]
Vnuc [[-96.95990904+2.22999621e-20j]]
Coulomb [[12.18381731-4.17663137e-20j]]
Exchange [[6.09190865-4.83100515e-10j]]
Kramer pair overlap [[-5.05745529e-10-4.82098289e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.07120708-1.53516506e-08j]]
Kinetic [[0.64300226+3.99773086e-10j]]
Vnuc [[-96.95990904-2.22999621e-20j]]
Coulomb [[12.18381731+4.17663137e-20j]]
Exchange [[6.09190865+4.83100515e-10j]]
Kramer pair overlap [[-5.05745529e-10+4.82098289e-10j]]
Orbital: 0    Norm: [0.99825117]    Update: [0.02257846]    Energy:[-45.00976846+3.50481306e-10j]
Scaled energy (4C Dirac energy) [-43.2232942]
=============Iteration: 6
F= [[-45.07120708-1.53516506e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-45.07120791+5.45900219e-10j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.05531918+8.05895853e-10j]]
Kinetic [[0.63929761-1.94295089e-11j]]
Vnuc [[-97.67750841+8.16792465e-19j]]
Coulomb [[12.27859141+1.61956794e-20j]]
Exchange [[6.1392957-7.58196437e-10j]]
Kramer pair overlap [[-5.24070937e-10-4.98400431e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.05531943-1.60109772e-08j]]
Kinetic [[0.63929761+4.00646248e-10j]]
Vnuc [[-97.67750841-8.16792465e-19j]]
Coulomb [[12.27859141-1.61956794e-20j]]
Exchange [[6.1392957+7.58196437e-10j]]
Kramer pair overlap [[-5.24070937e-10+4.98400431e-10j]]
Orbital: 0    Norm: [1.00002763]    Update: [0.00697313]    Energy:[-45.07120791+5.45900219e-10j]
Scaled energy (4C Dirac energy) [-43.27995021]
=============Iteration: 7
F= [[-45.05531943-1.60109772e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-45.05532023+8.2223127e-10j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.05541823+1.10227821e-09j]]
Kinetic [[0.63931997-1.39424943e-11j]]
Vnuc [[-97.69435753+9.04571631e-20j]]
Coulomb [[12.27921944-4.06386597e-20j]]
Exchange [[6.13960972-1.05322637e-09j]]
Kramer pair overlap [[-5.28870002e-10-5.02270672e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.05541849-1.64168766e-08j]]
Kinetic [[0.63931997+3.97607728e-10j]]
Vnuc [[-97.69435753-9.04571631e-20j]]
Coulomb [[12.27921944+4.06386597e-20j]]
Exchange [[6.13960972+1.05322637e-09j]]
Kramer pair overlap [[-5.28870002e-10+5.02270672e-10j]]
Orbital: 0    Norm: [1.00000058]    Update: [0.00024046]    Energy:[-45.05532023+8.2223127e-10j]
Scaled energy (4C Dirac energy) [-43.26530008]
=============Iteration: 8
F= [[-45.05541849-1.64168766e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-45.05541928+1.11916921e-09j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.05534558+1.43190403e-09j]]
Kinetic [[0.63930643-4.74719364e-12j]]
Vnuc [[-97.70219967-1.29372392e-19j]]
Coulomb [[12.27985518+2.77595093e-20j]]
Exchange [[6.13992758-1.38234792e-09j]]
Kramer pair overlap [[-5.3172031e-10-5.04539047e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.05534584-1.68109117e-08j]]
Kinetic [[0.63930643+3.89989972e-10j]]
Vnuc [[-97.70219967+1.29372392e-19j]]
Coulomb [[12.27985518-2.77595093e-20j]]
Exchange [[6.13992758+1.38234792e-09j]]
Kramer pair overlap [[-5.3172031e-10+5.04539047e-10j]]
Orbital: 0    Norm: [1.]    Update: [6.10887605e-05]    Energy:[-45.05541928+1.11916921e-09j]
Scaled energy (4C Dirac energy) [-43.26539141]
=============Iteration: 9
F= [[-45.05534584-1.68109117e-08j]]
S= [[1.+0.j]]
orthonorm
Post-Ortho S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
ExpandSol start S= [[1.+0.j]]
Shell energy [-45.05534663+1.44911701e-09j]
orthonorm
Post-Ortho S= [[1.+0.j]]
ExpandSol S= [[1.+0.j]]
ExpandSol final S= [[1.+0.j]]
orb1 Is rho zero? (should be) False False
Fock [[-45.0553443+1.72080459e-09j]]
Kinetic [[0.63930589+6.8082699e-12j]]
Vnuc [[-97.70252134+1.55425479e-19j]]
Coulomb [[12.27987696-8.73032268e-20j]]
Exchange [[6.13993847-1.67139984e-09j]]
Kramer pair overlap [[-5.32786777e-10-5.05373381e-10j]]
end ExpandSol
orb1 Is rho zero? (should be) False False
Fock [[-45.05534456-1.71238684e-08j]]
Kinetic [[0.63930589+3.79090059e-10j]]
Vnuc [[-97.70252134-1.55425479e-19j]]
Coulomb [[12.27987696+8.73032268e-20j]]
Exchange [[6.13993847+1.67139984e-09j]]
Kramer pair overlap [[-5.32786777e-10+5.05373381e-10j]]
Orbital: 0    Norm: [1.]    Update: [3.98930179e-06]    Energy:[-45.05534663+1.44911701e-09j]
Scaled energy (4C Dirac energy) [-43.26532442]